x-pack/filebeat/input/s3/input.go

```diff
@@ -372,15 +375,49 @@ func (p *s3Input) handleS3Objects(svc s3iface.ClientAPI, s3Infos []s3Info, errC
 		objectHash := s3ObjectHash(s3Info)
 
 		// read from s3 object
-		reader, err := p.newS3BucketReader(svc, s3Info)
+		resp, err := p.getS3ObjectResponse(svc, s3Info)
 		if err != nil {
 			err = errors.Wrap(err, "newS3BucketReader failed")
+			p.logger.Error(err)
 			s3Context.setError(err)
 			return err
 		}
 
-		if reader == nil {
-			continue
+		if resp == nil {
+			resp.Body.Close()
```

Comment:
I don't think that resp can be nil if there is no error?

Reply:
You are right! Will remove this.